### PR TITLE
add typing

### PR DESCRIPTION
**Pull request by dark0ghost**
# Description

Added type annotations to `aiogram.types.base` module

## Type of change

Please delete options that are not relevant.

- [x] Documentation (typos, code examples or any documentation update)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test A
- [ ] Test B

**Test Configuration**:
* Operating System: 
* Python version: 

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by dark0ghost** [comment]
add typing for function

File: aiogram/types/base.py [hunk 1]
```diff
@@ -30,7 +31,7 @@ class MetaTelegramObject(type):
     """
     _objects = {}
 
-    def __new__(mcs, name, bases, namespace, **kwargs):
+    def __new__(mcs, name: str, bases, namespace, **kwargs) -> T:
```
Comment (JrooTJunior) [inline aiogram/types/base.py]:
```suggestion
    def __new__(mcs: typing.Type[T], name: str, bases: typing.Tuple[typing.Type], namespace: typing.Dict[str, typing.Any], **kwargs: typing.Any) -> T:
```

File: aiogram/types/base.py [hunk 2]
```diff
@@ -71,7 +72,7 @@ class TelegramObject(ContextInstanceMixin, metaclass=MetaTelegramObject):
     Abstract class for telegram objects
     """
 
-    def __init__(self, conf=None, **kwargs):
+    def __init__(self, conf: typing.Dict[str, typing.Any]=None, **kwargs) -> None:
```
Comment (JrooTJunior) [inline aiogram/types/base.py]:
```suggestion
    def __init__(self, conf: typing.Optional[typing.Dict[str, typing.Any]]=None, **kwargs: typing.Any) -> None:
```

File: aiogram/types/base.py [hunk 3]
```diff
@@ -117,7 +118,7 @@ def props_aliases(self) -> typing.Dict[str, str]:
         return getattr(self, ALIASES_ATTR_NAME, {})
 
     @property
-    def values(self):
+    def values(self) -> T:
```
Comment (JrooTJunior) [inline aiogram/types/base.py]: values is not accessible via this property (deprecated)
That's mean need to mark this method as deprecated.

About typing:
```suggestion
    def values(self) -> typing.Tuple[str]:
```

File: aiogram/types/base.py [hunk 4]
```diff
@@ -188,7 +189,7 @@ def as_json(self) -> str:
         return json.dumps(self.to_python())
 
     @classmethod
-    def create(cls, *args, **kwargs):
+    def create(cls, *args, **kwargs) -> None:
```
Comment (JrooTJunior) [inline aiogram/types/base.py]:
```suggestion
    def create(cls, *args: typing.Any, **kwargs: typing.Any) -> None:
```

File: aiogram/types/base.py [hunk 5]
```diff
@@ -210,7 +211,7 @@ def __getitem__(self, item):
             return self.props[item].get_value(self)
         raise KeyError(item)
 
-    def __setitem__(self, key, value):
+    def __setitem__(self, key, value) -> None:
```
Comment (JrooTJunior) [inline aiogram/types/base.py]:
```suggestion
    def __setitem__(self, key: str, value: typing.Any) -> None:
```

File: aiogram/types/base.py [hunk 6]
```diff
@@ -222,7 +223,7 @@ def __setitem__(self, key, value):
             return self.props[key].set_value(self, value, self.conf.get('parent', None))
         raise KeyError(key)
 
-    def __contains__(self, item):
+    def __contains__(self, item) -> bool:
```
Comment (JrooTJunior) [inline aiogram/types/base.py]:
```suggestion
    def __contains__(self, item: str) -> bool:
```

File: aiogram/types/base.py [hunk 7]
```diff
@@ -232,7 +233,7 @@ def __contains__(self, item):
         self.clean()
         return item in self.values
 
-    def __iter__(self):
+    def __iter__(self) -> T:
```
Comment (JrooTJunior) [inline aiogram/types/base.py]: I think here is needed to use `typing.Iterator[str]` but need to check it

File: aiogram/types/base.py [hunk 8]
```diff
@@ -250,7 +251,7 @@ def iter_keys(self):
         for key, _ in self:
             yield key
 
-    def iter_values(self):
+    def iter_values(self) -> typing.Any:
```
Comment (JrooTJunior) [inline aiogram/types/base.py]:
```suggestion
    def iter_values(self) -> typing.Generator[typing.Any, None, None]:
```

File: aiogram/types/base.py [hunk 9]
```diff
@@ -259,7 +260,7 @@ def iter_values(self):
         for _, value in self:
             yield value
 
-    def __hash__(self):
+    def __hash__(self) -> str:
```
Comment (JrooTJunior) [inline aiogram/types/base.py]: Hash must always return an integer (read the python docs)

```suggestion
    def __hash__(self) -> int:
```

File: aiogram/types/base.py [hunk 10]
```diff
@@ -281,5 +282,5 @@ def _hash(obj):
 
         return result
 
-    def __eq__(self, other):
+    def __eq__(self, other) -> bool:
```
Comment (JrooTJunior) [inline aiogram/types/base.py]:
```suggestion
    def __eq__(self, other: TelegramObject) -> bool:
```

File: aiogram/types/base.py [hunk 11]
```diff
@@ -128,11 +129,11 @@ def values(self):
         return getattr(self, VALUES_ATTR_NAME)
 
     @property
-    def telegram_types(self):
+    def telegram_types(self) -> T:
```
Comment (JrooTJunior) [inline aiogram/types/base.py]:
```suggestion
    def telegram_types(self) -> typing.List[TelegramObject]:
```

File: aiogram/types/base.py [hunk 12]
```diff
@@ -128,11 +129,11 @@ def values(self):
...
         return type(self).telegram_types
 
     @classmethod
-    def to_object(cls, data):
+    def to_object(cls, data) -> T:
```
Comment (JrooTJunior) [inline aiogram/types/base.py]:
```suggestion
    def to_object(cls: typing.Type[T], data: typing.Dict[str, typing.Any]) -> T:
```

File: aiogram/types/base.py [hunk 13]
```diff
@@ -142,7 +143,7 @@ def to_object(cls, data):
         return cls(**data)
 
     @property
-    def bot(self):
+    def bot(self) -> T:
```
Comment (JrooTJunior) [inline aiogram/types/base.py]:
```suggestion
    def bot(self) -> Bot:
```

File: aiogram/types/base.py [hunk 14]
```diff
@@ -199,7 +200,7 @@ def __str__(self) -> str:
         """
         return self.as_json()
 
-    def __getitem__(self, item):
+    def __getitem__(self, item: typing.Union[str,int]) -> T:
```
Comment (JrooTJunior) [inline aiogram/types/base.py]:
```suggestion
    def __getitem__(self, item: typing.Union[str,int]) -> typing.Any:
```

File: aiogram/types/base.py (hunk 10 above)
Comment (dark0ghost) [reply aiogram/types/base.py]: check

File: aiogram/types/base.py [hunk 15]
```diff
@@ -210,7 +213,7 @@ def __getitem__(self, item):
             return self.props[item].get_value(self)
         raise KeyError(item)
 
-    def __setitem__(self, key, value):
+     def __setitem__(self, key: str, value: typing.Any) -> None:
```
Comment (JrooTJunior) [inline aiogram/types/base.py]:
```suggestion
    def __setitem__(self, key: str, value: typing.Any) -> None:
```

File: aiogram/types/base.py [hunk 16]
```diff
@@ -241,7 +244,7 @@ def __iter__(self):
         for item in self.to_python().items():
             yield item
 
-    def iter_keys(self):
+    def iter_keys(self) -> T:
```
Comment (JrooTJunior) [inline aiogram/types/base.py]: Why here you use T? iter_keys is generator so use generator/iterator annotation like iter_values

**Comment by dark0ghost** [comment]
py37 installed: aiohttp==3.6.2,aiohttp-socks==0.2.2,aioredis==1.3.0,alabaster==0.7.12,aresponses==1.1.1,async-timeout==3.0.1,atomicwrites==1.3.0,attrs==19.3.0,Babel==2.7.0,certifi==2019.9.11,chardet==3.0.4,coverage==4.5.3,docutils==0.15.2,emoji==0.5.4,filelock==3.0.12,hiredis==1.0.0,idna==2.8,imagesize==1.1.0,importlib-metadata==0.23,Jinja2==2.10.3,MarkupSafe==1.1.1,more-itertools==7.2.0,multidict==4.5.2,packaging==19.2,pluggy==0.13.0,py==1.8.0,Pygments==2.4.2,pyparsing==2.4.2,pytest==4.5.0,pytest-asyncio==0.10.0,python-rapidjson==0.8.0,pytz==2019.3,requests==2.22.0,rethinkdb==2.4.4,six==1.12.0,snowballstemmer==2.0.0,Sphinx==2.2.0,sphinx-rtd-theme==0.4.3,sphinxcontrib-applehelp==1.0.1,sphinxcontrib-devhelp==1.0.1,sphinxcontrib-htmlhelp==1.0.2,sphinxcontrib-jsmath==1.0.1,sphinxcontrib-programoutput==0.15,sphinxcontrib-qthelp==1.0.2,sphinxcontrib-serializinghtml==1.1.3,toml==0.10.0,tox==3.14.0,ujson==1.35,urllib3==1.25.6,uvloop==0.13.0,virtualenv==16.7.6,wcwidth==0.1.7,yarl==1.3.0,zipp==0.6.0
py37 run-test-pre: PYTHONHASHSEED='2189426222'
py37 run-test: commands[0] | pytest
============================================================================== test session starts ===============================================================================
platform linux -- Python 3.7.3, pytest-4.5.0, py-1.8.0, pluggy-0.13.0
cachedir: .tox/py37/.pytest_cache
rootdir: /home/egor/Загрузки/aiogram/aiogram
plugins: asyncio-0.10.0, aresponses-1.1.1
collected 283 items                                                                                                                                                              

tests/test_bot.py ........................................                                                                                                                 [ 14%]
tests/test_dispatcher.py .....                                                                                                                                             [ 15%]
tests/test_filters.py .................................................................................................................................................... [ 68%]
..............                                                                                                                                                             [ 73%] 
tests/test_message.py .                                                                                                                                                    [ 73%]
tests/test_states_group.py ........                                                                                                                                        [ 76%]
tests/test_bot/test_api.py ..                                                                                                                                              [ 77%]
tests/test_dispatcher/test_filters/test_builtin.py ....                                                                                                                    [ 78%]
tests/test_dispatcher/test_filters/test_state.py .                                                                                                                         [ 78%]
tests/test_utils/test_auth_widget.py .....                                                                                                                                 [ 80%]
tests/test_utils/test_helper.py ..                                                                                                                                         [ 81%]
tests/types/test_animation.py ......                                                                                                                                       [ 83%]
tests/types/test_chat.py .......                                                                                                                                           [ 85%]
tests/types/test_chat_member.py ........                                                                                                                                   [ 88%]
tests/types/test_document.py ......                                                                                                                                        [ 90%]
tests/types/test_game.py .....                                                                                                                                             [ 92%]
tests/types/test_message.py ......                                                                                                                                         [ 94%]
tests/types/test_photo.py ....                                                                                                                                             [ 96%]
tests/types/test_update.py ...                                                                                                                                             [ 97%]
tests/types/test_user.py ........                                                                                                                                          [100%]

================================================================================ warnings summary ================================================================================
tests/test_filters.py::TestTextFilter::test_endswith[-exAmple_string-False]
  /home/egor/Загрузки/aiogram/aiogram/aiogram/dispatcher/filters/builtin.py:295: RuntimeWarning: coroutine 'BaseBot.close' was never awaited
    endswith = list(map(_pre_process_func, self.endswith))

tests/test_utils/test_auth_widget.py::test_generate_hash
  /home/egor/Загрузки/aiogram/aiogram/tests/test_utils/test_auth_widget.py:23: DeprecationWarning: Call to deprecated function generate_hash (`generate_hash` is outdated, please use `check_signature` or `check_integrity`).
    res = generate_hash(data, TOKEN)

tests/test_utils/test_auth_widget.py::Test_check_token::test_ok
  /home/egor/Загрузки/aiogram/aiogram/tests/test_utils/test_auth_widget.py:32: DeprecationWarning: Call to deprecated function check_token (`check_token` helper was renamed to `check_integrity`).
    assert check_token(data, TOKEN) is True

tests/test_utils/test_auth_widget.py::Test_check_token::test_ok
tests/test_utils/test_auth_widget.py::Test_check_token::test_fail
  /home/egor/Загрузки/aiogram/aiogram/aiogram/utils/auth_widget.py:40: DeprecationWarning: Call to deprecated function generate_hash (`generate_hash` is outdated, please use `check_signature` or `check_integrity`).
    return param_hash == generate_hash(data, token)

tests/test_utils/test_auth_widget.py::Test_check_token::test_fail
  /home/egor/Загрузки/aiogram/aiogram/tests/test_utils/test_auth_widget.py:36: DeprecationWarning: Call to deprecated function check_token (`check_token` helper was renamed to `check_integrity`).
    assert check_token(data, TOKEN) is False

-- Docs: https://docs.pytest.org/en/latest/warnings.html
===================================================================== 283 passed, 6 warnings in 0.70 seconds =====================================================================
____________________________________________________________________________________ summary _____________________________________________________________________________________
  py37: commands succeeded
  congratulations :)

**Comment by JrooTJunior** [comment]
Thanks for the work and patience

**Pull request merged**